frontend/packages/console-app/src/components/access-modes/access-mode.tsx

```diff
@@ -26,62 +25,64 @@ export const getPVCAccessModes = (resource: PersistentVolumeClaimKind, key: stri
 
 export const AccessModeSelector: React.FC<AccessModeSelectorProps> = (props) => {
   const {
-    formClassName,
+    className,
     pvcResource,
+    filterByVolumeMode,
     onChange,
     loaded,
     provisioner,
     availableAccessModes = [],
+    description,
+    ignoreReadOnly,
   } = props;
 
   const { t } = useTranslation();
-
   const pvcInitialAccessMode = pvcResource
     ? getPVCAccessModes(pvcResource, 'value')
     : availableAccessModes;
 
   const [allowedAccessModes, setAllowedAccessModes] = React.useState<string[]>();
-  const [accessMode, setAccessMode] = React.useState<string>('');
+  const [accessMode, setAccessMode] = React.useState<string>();
 
-  const updateAllowedAccessModes = (scProvisioner: string) =>
-    provisionerAccessModeMapping[scProvisioner] ?? getAccessModeForProvisioner('');
+  const changeAccessMode = React.useCallback(
+    (mode: string) => {
+      setAccessMode(mode);
+      onChange(mode);
+    },
+    [onChange],
+  );
 
   React.useEffect(() => {
     if (loaded) {
-      let currentModes: string[];
-      provisioner?.includes(cephRBDProvisionerSuffix) && pvcResource?.spec.volumeMode !== 'Block'
-        ? (currentModes = ['ReadWriteOnce', 'ReadOnlyMany'])
-        : (currentModes = updateAllowedAccessModes(provisioner));
-      setAllowedAccessModes(currentModes);
+      setAllowedAccessModes(
+        getAccessModeForProvisioner(
+          provisioner,
+          ignoreReadOnly,
+          filterByVolumeMode ? pvcResource?.spec?.volumeMode : null,
+        ),
+      );
     }
-  }, [loaded, provisioner, pvcResource]);
+  }, [filterByVolumeMode, ignoreReadOnly, loaded, provisioner, pvcResource]);
```

Comment:
It would be better if we used `pvcResource?.spec?.volumeMode` as the dependency instead of `pcvResource`. That way the hook isn't reevaluated if other `pvcResource` values change, only the one we're using.

Reply:
ack